--- GMT4.3.0/configure.nostrip	2008-05-02 16:38:12.000000000 -0600
+++ GMT4.3.0/configure	2008-05-06 10:56:49.000000000 -0600
@@ -11114,11 +11114,10 @@
 	CPPFLAGS="$CPPFLAGS -DDEBUG"
 	CFLAGS="$CFLAGS -g"
 else
-	CFLAGS=`echo $CFLAGS | sed 's/-g//'`
 	if test "$os" = "Darwin" ; then
 		LDFLAGS="$LDFLAGS -Wl,-x"
 	else
-		LDFLAGS="$LDFLAGS -s"
+		LDFLAGS="$LDFLAGS"
 	fi
 	if test "$GCC" = "yes" || test "$os" = "Linux" || test "$os" = "AIX"; then
 		CFLAGS="$CFLAGS -O2"
