--- GMT4.3.0/src/makegmt.macros.in.destdir	2008-04-30 01:52:33.000000000 -0600
+++ GMT4.3.0/src/makegmt.macros.in	2008-05-05 10:51:32.000000000 -0600
@@ -76,8 +76,8 @@
 MEX_LIB	= @MEX_LIB@
 MEX_INC	= @MEX_INC@
 MEX_EXT	= @MEX_EXT@
-MEX_XDIR	= @MEX_XDIR@
-MEX_MDIR	= @MEX_MDIR@
+MEX_XDIR	= $(DESTDIR)@MEX_XDIR@
+MEX_MDIR	= $(DESTDIR)@MEX_MDIR@
 #
 #-------------------------------------------------------------------------------
 #	Miscellaneous macros for Cygwin
