--- GMT4.3.0/src/x2sys/makefile.shlibs	2008-04-25 14:14:36.000000000 -0600
+++ GMT4.3.0/src/x2sys/makefile	2008-05-12 13:39:35.000000000 -0600
@@ -11,7 +11,7 @@
 include $(GMTSRCDIR)gmtalldeps.macros
 
 INCLUDES	= -I$(srcdir) -I../mgg -I../mgd77 -I$(NETCDF_INC)
-ALLLIB		= -L../mgd77 -lmgd77 -L../mgg -lgmt_mgg -L.. -lgmt $(NC_LIBS) $(LIBS)
+ALLLIB		= -L../mgd77 -lmgd77 -L../mgg -lgmt_mgg -L.. -lgmt $(LIBS)
 
 PROGS_H		= x2sys.h
 PROGS_O		= x2sys_init.o x2sys_put.o x2sys_get.o x2sys_binlist.o x2sys_datalist.o x2sys_cross.o
--- GMT4.3.0/src/makefile.shlibs	2008-05-02 16:37:06.000000000 -0600
+++ GMT4.3.0/src/makefile	2008-05-12 13:42:22.000000000 -0600
@@ -198,13 +198,13 @@
 		$(RANLIB) $@
 
 libpsl.$(SL):	libpsl.a
-		$(LD) $(LD_OPT) pslib.o $(NC_LIBS) $(LIBS) -o $@
+		$(LD) $(LD_OPT) pslib.o $(LIBS) -o $@
 
 libgmt.$(SL):	libgmt.a
 		$(LD) $(LD_OPT) $(LIB_O) $(TRIANGLE_O) $(ALPHA_SINCOS_O) $(NC_LIBS) $(LIBS) -o $@
 
 libgmtps.$(SL):	libgmtps.a
-		$(LD) $(LD_OPT) $(LIBPS_O) libpsl.$(SL) $(NC_LIBS) $(LIBS) -o $@
+		$(LD) $(LD_OPT) $(LIBPS_O) libpsl.$(SL) libgmt.$(SL) $(LIBS) -o $@
 
 #-------------------------------------------------------------------------------
 #	program rules
