diff -up gmt-5.1.2/share/tools/CMakeLists.txt.bash-completion gmt-5.1.2/share/tools/CMakeLists.txt
--- gmt-5.1.2/share/tools/CMakeLists.txt.bash-completion	2014-12-28 14:46:54.000000000 -0700
+++ gmt-5.1.2/share/tools/CMakeLists.txt	2015-07-28 12:15:07.627831526 -0600
@@ -29,6 +29,9 @@ install (PROGRAMS
 	DESTINATION ${GMT_DATADIR}/tools
 	COMPONENT Runtime)
 
+option(BASH_COMPLETION_DIR "Install location for gmt bash completion file"
+etc/bash_completion.d)
+
 if (GMT_INSTALL_TRADITIONAL_FOLDERNAMES)
 	install (PROGRAMS
 		gmt_completion.bash
@@ -37,7 +40,7 @@ if (GMT_INSTALL_TRADITIONAL_FOLDERNAMES)
 else (GMT_INSTALL_TRADITIONAL_FOLDERNAMES)
 	install (FILES
 		gmt_completion.bash
-		DESTINATION etc/bash_completion.d
+		DESTINATION ${BASH_COMPLETION_DIR}
 		COMPONENT Runtime
 		RENAME gmt)
 endif (GMT_INSTALL_TRADITIONAL_FOLDERNAMES)
